src/goTest.ts

```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+import cp = require('child_process');
+import path = require('path');
+import vscode = require('vscode');
+import util = require('util');
+import { getGoRuntimePath } from './goPath'
+
+/**
+* Executes the unit test at the primary cursor using `go test`. Output
+* is sent to the 'Go' channel.
+* 
+* @param timeout an optional ParseDuration formatted timeout string for the tests.
+*
+* TODO: go test returns filenames with no path information for failures,
+* so output doesn't produce navigable line references.
+*/
+export function testAtCursor(timeout: string) {
+	var editor = vscode.window.activeTextEditor;
+	if (!editor) {
+		vscode.window.showInformationMessage("No editor is active.");
+		return;
+	}
+	if (!timeout) timeout = '30s';
```

Comment:
Curious - is this necessary?  I would have expected the default set in package.json would be provided in the configuration value you get back.